Tests/OptimizelyTests-APIs/OptimizelyClientTests_Others.swift

```diff
@@ -399,5 +399,28 @@ class OptimizelyClientTests_Others: XCTestCase {
         XCTAssertEqual(newHandlersCount, oldHandlersCount! - 1, "nil handlers should be filtered out")
     }
     
+    func testGettingLoggerAfterMultiInit() {
+        
+        let exp = expectation(description: "a")
 
+        var optimizely = OptimizelyClient(sdkKey: "a")
+         for i in 0..<1000 {
+             DispatchQueue.global().async {
+                 if i == 10 {
+                     optimizely = OptimizelyClient(sdkKey: "b")
+                 }
+                 for k in 0..<10 {
+                     let logger = OPTLoggerFactory.getLogger()
+                     logger.log(level: .info, message: "[LOGGER] [\(i)] \(k)")
+                 }
+                if i == 999 {
+                    exp.fulfill()
+                }
+            }
+            // need to let the main global queue run otherwise some remained queued.
+            Thread.sleep(forTimeInterval: 0.02)
```

Comment:
This sleep delay breaks the test. 1000 async tasks not concurrent any more because this delay.
If we remove the sleep delay and using DispatchGroup for sync, this test fails because concurrency issue.

Reply:
> This sleep delay breaks the test. 1000 async tasks not concurrent any more because this delay.
> If we remove the sleep delay and using DispatchGroup for sync, this test fails because concurrency issue.

Please add a test if you think it would break.  The test won't run all the queued in the global queue without the sleep.  So.....